app/src/main/java/com/medmapper/v33001/dto/Medicine.kt

```diff
@@ -9,17 +11,21 @@ data class Medicine(
     val quantity: Int = 0,
     val prescriptionStrength: String = "",
     val startDate: LocalDate = LocalDate.now(),
-    val prescriptionLength: String = "",
-    val time: Long = 0,
-    // Frequency per 24 hours
+    val prescriptionLength: Duration = Duration.ZERO,
+    val timeUnit: TimeUnit = TimeUnit.HOURS,
     val frequency: Int = 0) {
     override fun toString(): String {
         return "$name, $quantity , $prescriptionStrength"
     }
 
-    // Need to properly create function or service to translate string to dates, etc.
-    /*fun getEndDate(): String {
-        var endDate = startDate + prescriptionLength
-        return endDate
-    }*/
-}
\ No newline at end of file
+    fun getEndDate(): LocalDate {
+        return startDate.plus(prescriptionLength)
+    }
+}
```

Comment:
Use the LocalDate.plus() method to add the prescriptionLength to the startDate and get the end date of the prescription.

Source: https://docs.oracle.com/javase/8/docs/api/java/time/LocalDateTime.html

Reply:
Good.